### PR TITLE
Enable .define blah and ~blah to use images

### LINKED ISSUE #139: Allow .define to work on images
If a definition contains nothing but an image URL, I think the image should be uploaded when the definition is invoked.

### PR DESCRIPTION
**Pull request by iandioch**
Closes #139.

I think it's an ok way of doing it.

Untested. Unrun. Unloved.

### CODE REVIEW

File: steely/plugins/define.py [hunk 1]
```diff
@@ -42,18 +44,39 @@ def main(bot, author_id, message, thread_id, thread_type, **kwargs):
...
         command = f'~{command}'
     if len(command) > LIMIT:
         bot.sendMessage("that's too long dickhead",
                         thread_id=thread_id, thread_type=thread_type)
         return
     search = CMD_DB.get(CMD.cmd == command)
+
+    data = {}
+    if image_url is not None:
+        data["image"] = image_url
```
Comment (devoxel) [inline steely/plugins/define.py]: So here are you removing the old text? What is the state of the db after creating a piece of text and then defining an image

File: steely/plugins/define.py [hunk 2]
```diff
@@ -42,18 +44,39 @@ def main(bot, author_id, message, thread_id, thread_type, **kwargs):
             bot.sendMessage(ANGRY_STRING, thread_id=thread_id,
                             thread_type=thread_type)
             return
+    elif command == 'image':
+        if ' ' in text:
+            command, image_url = text.split(' ', 1)
+            try:
+                bot.sendRemoteImage(
```
Comment (devoxel) [inline steely/plugins/define.py]: the nested logic here is doirty, could be worth moving everything to functions so it's clearer

File: steely/plugins/define.py (hunk 1 above)
Comment (iandioch) [reply steely/plugins/define.py]: ooh good question.

I think each ~word should only refer to either an image or a sentence or a code, just like overwriting words with code now deletes the words.

File: steely/plugins/define.py (hunk 2 above)
Comment (iandioch) [reply steely/plugins/define.py]: can do friend

**Comment by iandioch** [comment]
bump @devoxel

**Pull request merged**